db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1697,13 +1687,20 @@ Status DBImpl::ReFitLevel(ColumnFamilyData* cfd, int level, int target_level) {
 
     VersionEdit edit;
     edit.SetColumnFamily(cfd->GetID());
+
+    uint64_t l0_epoch_num = kUnknownL0EpochNumber;
+    if (to_level == 0) {
+      versions_->FetchAddL0EpochNumber(vstorage->LevelFiles(level).size());
+      l0_epoch_num = versions_->current_next_l0_epoch_number() - 1;
+    }
     for (const auto& f : vstorage->LevelFiles(level)) {
       edit.DeleteFile(level, f->fd.GetNumber());
       edit.AddFile(
           to_level, f->fd.GetNumber(), f->fd.GetPathId(), f->fd.GetFileSize(),
           f->smallest, f->largest, f->fd.smallest_seqno, f->fd.largest_seqno,
           f->marked_for_compaction, f->temperature, f->oldest_blob_file_number,
-          f->oldest_ancester_time, f->file_creation_time, f->file_checksum,
+          f->oldest_ancester_time, f->file_creation_time,
+          to_level == 0 ? l0_epoch_num-- : f->l0_epoch_number, f->file_checksum,
```

Comment:
I think it should be fine that we assign l0_epoch_num based on `vstorage->LevelFiles(level)`'s order (i.e, smallest key in non-zero level) since files should not have overlapped keys. Correct me if I am missing anything.